JSQMessagesViewController/Views/JSQMessagesTypingIndicatorFooterView.m

```diff
@@ -109,7 +124,123 @@ - (void)configureWithEllipsisColor:(UIColor *)ellipsisColor
     
     [self setNeedsUpdateConstraints];
     
-    self.typingIndicatorImageView.image = [[UIImage jsq_defaultTypingIndicatorImage] jsq_imageMaskedWithColor:ellipsisColor];
+    // Configure correct color and (whether/not) animated
+    [self setTypingIndicatorAnimated:animated :ellipsisColor];
+}
+
+- (void) setTypingIndicatorEllipsisColor:(UIColor*) ellipsisColor
+{
+    self.typingIndicatorView1.backgroundColor = ellipsisColor;
+    self.typingIndicatorView2.backgroundColor = ellipsisColor;
+    self.typingIndicatorView3.backgroundColor = ellipsisColor;
+}
+
+- (void) setTypingIndicatorAnimated:(BOOL) animated :(UIColor*) ellipsisColor
+{
+    // Configure Base Color
+    [self setTypingIndicatorEllipsisColor:ellipsisColor];
+    // Remove existing animator if exist
+    if (self.animator) {
+        [self.animator stopAnimating];
+    }
+    // If animated -> New Animator
+    if (animated) {
+        self.animator = [[JSQMessagesTypingIndicatorAnimator alloc] initWithTypingIndicatorView:self :ellipsisColor];
+        [self.animator startAnimating];
+    }
+}
+
+@end
+
+@implementation JSQMessagesTypingIndicatorAnimator
+
+- (id) initWithTypingIndicatorView:(JSQMessagesTypingIndicatorFooterView *)view :(UIColor*) ellipsisColor
+{
+    if (self = [self init]) {
+        self.typingView = view;
+        self.ellipsisColor = ellipsisColor;
+        self.animating = NO;
+        
+        // Initialize Colors
+        self.typingView.typingIndicatorView1.backgroundColor = ellipsisColor;
+        self.typingView.typingIndicatorView2.backgroundColor = ellipsisColor;
+        self.typingView.typingIndicatorView3.backgroundColor = ellipsisColor;
+    }
+    return self;
+}
+
+- (void) startAnimating
+{
+    self.animating = YES;
+    [self startTypingIndicatorAnimating:self.ellipsisColor];
+}
+
+- (void) stopAnimating
+{
+    self.animating = NO;
+}
+
+// How long does each individual dot stay dark for?
+#define kTypingIndicatorSingleDot_Darker_AnimationDuration 1.5f
+// How long does it take to transition from one dark circle to another?
+#define kTypingIndicatorSingleDot_TransitionTime 0.4f
+// Darkening co-efficient
+#define kTypingIndicatorAnimatedDot_Darkening_Coefficient 3.0f
```

Comment:
no `#define`s

let's either: (1) define constants inline where these are used, or (2) change these to `static const CGFloat`